src/wing/CCPACSWings.cpp

```diff
@@ -141,18 +102,18 @@ bool CCPACSWings::HasWing(const std::string & uid) const
     return false;
 }
 
-void CCPACSWings::AddWing(CCPACSWing* wing)
+void CCPACSWings::Add(CCPACSWing* wing)
 {
-    // Check whether the same wing already exists if yes remove it before adding the new one
+    m_wings.push_back(unique_ptr<CCPACSWing>(wing));
+}
+
+void CCPACSWings::Remove(CCPACSWing* wing) {
     for (std::vector<unique_ptr<CCPACSWing> >::iterator it = m_wings.begin(); it != m_wings.end(); ++it) {
-        if ((*it)->GetUID() == wing->GetUID()) {
+        if ((*it).get() == wing) {
             m_wings.erase(it);
-            break;
+            return;
         }
     }
-
-    // Add the new wing to the wing list
-    m_wings.push_back(unique_ptr<CCPACSWing>(wing));
+    throw std::runtime_error("Wing not found");
```

Comment:
I agree with you, that CTiglError should inherit from std::exception. This should be done separately and also has no priority.